### PR TITLE
Develop nutty modify framework addlayoutparameterclone

### PR DESCRIPTION
**Pull request by nutty898**
1. Fixed button scale bugs;
2. Fixed bugs of GUIReader;
3. Add clone method to Layoutparameter;
4. Fixed bugs of label;
5. Optimize Layout clipping;

### CODE REVIEW

File: cocos/gui/UIButton.cpp [hunk 1]
```diff
@@ -366,8 +366,7 @@ void Button::onPressStateChangedToNormal()
     else
     {
         _buttonNormalRenderer->stopAllActions();
-        Action *zoomAction = ScaleTo::create(0.05f, _normalTextureScaleXInSize, _normalTextureScaleYInSize);
-        _buttonNormalRenderer->runAction(zoomAction);
+        _buttonNormalRenderer->setScale(_normalTextureScaleXInSize, _normalTextureScaleYInSize);
```
Comment (dumganhar) [inline cocos/gui/UIButton.cpp:369]: Why not use `Action` now?

File: cocos/gui/UILayoutParameter.h [hunk 2]
```diff
@@ -130,6 +134,8 @@ class LinearLayoutParameter : public LayoutParameter
      * @return LinearGravity
      */
     LinearGravity getGravity() const;
+    virtual LayoutParameter* createCloneInstance();
```
Comment (dumganhar) [inline cocos/gui/UILayoutParameter.h]: Please add `override` keyword for override functions.

Comment (nutty898) [inline cocos/gui/UILayoutParameter.h]: ok

File: cocos/gui/UIButton.cpp (hunk 1 above)
Comment (nutty898) [inline cocos/gui/UIButton.cpp:369]: It will cause scale bugs. So don't use Action temporarily.

Comment (dumganhar) [inline cocos/gui/UIButton.cpp:369]: What's the bug? Could you describe it? Maybe we could help you.

**Comment by dumganhar** [comment]
Could you please add the detail changes of this PR?
Thanks.

File: cocos/gui/UILayout.cpp [hunk 3]
```diff
@@ -421,79 +424,83 @@ void Layout::setStencilClippingSize(const Size &size)
...
 {
-    Point worldPos = convertToWorldSpace(Point::ZERO);
-    AffineTransform t = getNodeToWorldAffineTransform();
-    float scissorWidth = _size.width*t.a;
-    float scissorHeight = _size.height*t.d;
-    Rect parentClippingRect;
-    Layout* parent = this;
-    bool firstClippingParentFounded = false;
-    while (parent)
+    if (_clippingRectDirty)
```
Comment (zilongshanren) [inline cocos/gui/UILayout.cpp:428]: could you add another function to wrap all of these code?
This function is very complex for me to grasp

**Comment by minggo** [comment]
Please break it into more small pull requests in future.
Thanks.

**Comment by minggo** [comment]
Each pull request for an issue.

**Pull request merged**